app/src/main/java/org/listenbrainz/android/ui/screens/brainzplayer/BrainzPlayerScreen.kt

```diff
@@ -50,19 +57,28 @@ fun BrainzPlayerScreen() {
     val songsViewModel = hiltViewModel<SongViewModel>()
     val artistViewModel = hiltViewModel<ArtistViewModel>()
     val playlistViewModel = hiltViewModel<PlaylistViewModel>()
+    val brainzPlayerViewModel = hiltViewModel<BrainzPlayerViewModel>()
     
     // Data streams
     val albums = albumViewModel.albums.collectAsState(initial = listOf()).value     // TODO: Introduce initial values to avoid flicker.
     val songs = songsViewModel.songs.collectAsState(initial = listOf()).value
     val artists = artistViewModel.artists.collectAsState(initial = listOf()).value
     val playlists by playlistViewModel.playlists.collectAsState(initial = listOf())
-    val recentlyPlayed = Playlist.recentlyPlayed
+    val songsPlayedToday = brainzPlayerViewModel.songsPlayedToday.collectAsState(initial = listOf()).value
+    val recentlyPlayed = brainzPlayerViewModel.recentlyPlayed.collectAsState(initial = mutableListOf()).value
+    val topRecents = recentlyPlayed.subList(0, minOf(recentlyPlayed.size , 5)).toMutableList()
```

Comment:
Why use 5 here? This is crashing the app for me because this causes index out of bounds exception

Reply:
@akshaaatt 
This 5 is used to get the top 5 recently played songs, displayed on the overview screen. After the 5th one, the users see all recently played options, and on clicking that, they are redirected to the recent plays screen. This shouldn't be causing a crash tho because I used minof so that if the list size is less than 5, that would be returned.
I had tested this once with a list whose size was less than 5 and it had worked well, Ill just test it again and let u know. 
Thanks!

Reply:
@akshaaatt, I retested with an empty list, and it works fine.. no crashes. I don't think theres an issue due to this..

Reply:
`FATAL EXCEPTION: main
                                                                                                    Process: org.listenbrainz.android.debug, PID: 20789
                                                                                                    java.lang.IndexOutOfBoundsException: toIndex = 5
                                                                                                    	at java.util.AbstractList.subListRangeCheck(AbstractList.java:507)
                                                                                                    	at java.util.ArrayList.subList(ArrayList.java:1125)
                                                                                                    	at org.listenbrainz.android.ui.screens.brainzplayer.BrainzPlayerScreenKt.BrainzPlayerScreen(BrainzPlayerScreen.kt:71)
                                                                                                    	at org.listenbrainz.android.ui.screens.brainzplayer.BrainzPlayerScreenKt$BrainzPlayerScreen$2.invoke(Unknown Source:8)
                                                                                                    	at org.listenbrainz.android.ui.screens.brainzplayer.BrainzPlayerScreenKt$BrainzPlayerScreen$2.invoke(Unknown Source:10)
                                                                                                    	at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:192)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2556)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.skipCurrentGroup(Composer.kt:2827)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:3314)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.recompose$runtime_release(Composer.kt:3265)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.recompose(Composition.kt:940)
                                                                                                    	at androidx.compose.runtime.Recomposer.performRecompose(Recomposer.kt:1155)
                                                                                                    	at androidx.compose.runtime.Recomposer.access$performRecompose(Recomposer.kt:127)
                                                                                                    	at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$1.invoke(Recomposer.kt:583)
                                                                                                    	at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$1.invoke(Recomposer.kt:551)
                                                                                                    	at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame(AndroidUiFrameClock.android.kt:41)
                                                                                                    	at androidx.compose.ui.platform.AndroidUiDispatcher.performFrameDispatch(AndroidUiDispatcher.android.kt:109)
                                                                                                    	at androidx.compose.ui.platform.AndroidUiDispatcher.access$performFrameDispatch(AndroidUiDispatcher.android.kt:41)
                                                                                                    	at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame(AndroidUiDispatcher.android.kt:69)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
                                                                                                    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [androidx.compose.runtime.PausableMonotonicFrameClock@a09516d, androidx.compose.ui.platform.MotionDurationScaleImpl@41511a2, StandaloneCoroutine{Cancelling}@da40e33, AndroidUiDispatcher@99fc5f0]`

Reply:
`FATAL EXCEPTION: main
                                                                                                    Process: org.listenbrainz.android.debug, PID: 26816
                                                                                                    java.lang.IllegalArgumentException: fromIndex(0) > toIndex(-1)`

Reply:
Ahh I understood.. if list is completely empty, accessing index 0 is not possible. Will rectify it